### PR TITLE
Odata service#90

### LINKED ISSUE #90: Create Odata Response to Redfish
Create the response to /redfish/v1/odata URL with the offered services

### PR DESCRIPTION
**Pull request by ffdarkpenguin**
Closes #90

---
This change is [Reviewable](https://reviewable.io/reviews/hewlettpackard/oneview-redfish-toolkit/93)

### CODE REVIEW

**Comment by danielreed** [comment]
Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvD2SvZ9t3h09ml5gqq:-KvD2SvZ9t3h09ml5gqr:b-3rybta) (no related file):*
I'm confused about why you removed EventService and Sessions from ServiceRoot.  Don't they
also belong there?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93)*

**Comment by ffdarkpenguin** [comment]
Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvD2SvZ9t3h09ml5gqq:-KvE5L0Uunfez-CSojnP:bdgfe5l) (no related file):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

I'm confused about why you removed EventService and Sessions from ServiceRoot.  Don't they
also belong there?
</blockquote></details>

They do, but they are not implemented. So in order to the tool not fail I changed it to what is implemented only. When and if we add EventService we add it there and in the Odata. Does it make sense to you? And is it ok?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93)*

**Comment by danielreed** [comment]
:lgtm:

---

Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvD2SvZ9t3h09ml5gqq:-KvE67VJy3CH-Uc7hzDI:bg2j5hf) (no related file):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

They do, but they are not implemented. So in order to the tool not fail I changed it to what is implemented only. When and if we add EventService we add it there and in the Odata. Does it make sense to you? And is it ok?
</blockquote></details>

Makes sense, thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-:-KvE6AaZEtebJBY_cov6:bnfp4nl)*

**Comment by llvieira** [comment]
Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[oneview_redfish_toolkit/blueprints/odata.py, line 52 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvE6v4yiLLFIPdbqD6Y:-KvE6v4yiLLFIPdbqD6Z:b-p24ndh) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/d55a812dc6ded60fd95a59e761ecee5c9616de15/oneview_redfish_toolkit/blueprints/odata.py#L52)):*
> ```Python
> 
> @odata.errorhandler(status.HTTP_404_NOT_FOUND)
> def not_found(error):
> ```

This method it's not used, I think you can remove it, okay?

---

*[oneview_redfish_toolkit/tests/blueprints/test_odata.py, line 28 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvE7DQOwUy5zekX7r8S:-KvE7DQPS1r_EobhjnCd:b-gmqfd7) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/d55a812dc6ded60fd95a59e761ecee5c9616de15/oneview_redfish_toolkit/tests/blueprints/test_odata.py#L28)):*
> ```Python
> 
> 
> class TestOdata(unittest.TestCase):
> ```

Don't need to test Blueprint when it raise an exception?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93)*

File: run.py [hunk 1]
```diff
@@ -14,7 +14,49 @@
...
 from oneview_redfish_toolkit.app import app
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=5000, debug=True)
+    config = configparser.ConfigParser()
+    config.optionxform = str
+    try:
+        config.read('redfish.conf')
+    except Exception:
+        print("Failed to load config file. Aborting...")
```
Comment (llvieira) [inline run.py]: Use 'print' is good here?

**Comment by ffdarkpenguin** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[run.py, line 28 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-Kv9SVkmsbvKkxbtT_s--r1-28:-KvE8k2dskVEIU3mn-1N:b-t8fub0) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/d55a812dc6ded60fd95a59e761ecee5c9616de15/run.py#L28)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Use 'print' is good here?
</blockquote></details>

this is from the previous patch and you commented and I replied and we are still waiting on PO decision

---

*[oneview_redfish_toolkit/blueprints/odata.py, line 52 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvE6v4yiLLFIPdbqD6Y:-KvE9EU92kbLKaYSCikY:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/d55a812dc6ded60fd95a59e761ecee5c9616de15/oneview_redfish_toolkit/blueprints/odata.py#L52)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

This method it's not used, I think you can remove it, okay?
</blockquote></details>

Done.

---

*[oneview_redfish_toolkit/tests/blueprints/test_odata.py, line 28 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-KvE7DQOwUy5zekX7r8S:-KvE9RBAUhjbXVyJeg_5:b-xmltxn) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/d55a812dc6ded60fd95a59e761ecee5c9616de15/oneview_redfish_toolkit/tests/blueprints/test_odata.py#L28)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Don't need to test Blueprint when it raise an exception?
</blockquote></details>

the only possible way for this to fail is if the class instantiation fails and this was tested on API tests. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93)*

**Comment by llvieira** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-:-KvEAyuNp13AcHYk62BZ:bnfp4nl)*

**Comment by bsoutham** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/93#-:-KvEKS6_meqbmiygZCZ_:bnfp4nl)*

**Pull request merged**